compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -516,6 +516,27 @@ impl<'a> Parser<'a> {
             token::BinOp(token::And) | token::AndAnd => {
                 make_it!(this, attrs, |this, _| this.parse_borrow_expr(lo))
             }
+            token::BinOp(token::Plus) => {
+                let mut err = this.struct_span_err(lo, "leading `+` is not supported");
+                err.span_label(lo, "unexpected `+`");
```

Comment:
We might want to replace this with

```suggestion
                let mut err = this.unexpected().unwrap_err();
```

Reply:
I'm getting some unexpected results after changing this to unexpected():

```
error: expected `#`, found `+`
 --> src/test/ui/parser/issue-88276-unary-plus.rs:4:13
  |
4 |     let _ = +1; //~ ERROR leading `+` is not supported
  |             ^ expected `#`
  |
```